crates/sui-kvstore/src/bigtable/client.rs

```diff
@@ -0,0 +1,321 @@
+                        cell_value.append(&mut chunk.value);
+                    }
+                    None => {
+                        if chunk.timestamp_micros == 0 {
+                            cell_value.append(&mut chunk.value);
+                        } else if chunk.timestamp_micros >= timestamp {
+                            // newer version of cell is available
+                            timestamp = chunk.timestamp_micros;
+                            cell_value = chunk.value;
+                        }
+                    }
+                }
+                if chunk.row_status.is_some() {
+                    if let Some(RowStatus::CommitRow(_)) = chunk.row_status {
+                        if let Some(cell_name) = cell_name {
+                            row.push((cell_name, cell_value));
+                        }
+                        if let Some(row_key) = row_key {
+                            result.push((row_key, row));
+                        }
+                    }
+                    row_key = None;
+                    row = vec![];
+                    cell_value = vec![];
+                    cell_name = None;
+                }
+            }
+        }
+        Ok(result)
+    }
+
+    async fn multi_set<V: Serialize>(
+        &mut self,
+        table_name: &str,
+        values: impl IntoIterator<Item = (Vec<u8>, V)> + std::marker::Send,
+    ) -> Result<()> {
+        let mut entries = vec![];
+        for (key, value) in values {
+            let entry = Entry {
+                row_key: key,
+                mutations: vec![Mutation {
+                    mutation: Some(mutation::Mutation::SetCell(SetCell {
+                        family_name: COLUMN_FAMILY_NAME.to_string(),
+                        column_qualifier: COLUMN_QUALIFIER.to_owned().into_bytes(),
+                        timestamp_micros: -1,
+                        value: bcs::to_bytes(&value)?,
+                    })),
+                }],
+            };
+            entries.push(entry);
+        }
+        let request = MutateRowsRequest {
+            table_name: format!("{}{}", self.table_prefix, table_name),
+            entries,
+            ..MutateRowsRequest::default()
+        };
+        self.mutate_rows(request).await?;
+        Ok(())
+    }
+
+    pub async fn multi_get<V>(&mut self, table_name: &str, keys: Vec<Vec<u8>>) -> Result<Vec<V>>
+    where
+        V: DeserializeOwned,
+    {
+        let request = ReadRowsRequest {
+            table_name: format!("{}{}", self.table_prefix, table_name),
+            rows_limit: keys.len() as i64,
+            rows: Some(RowSet {
+                row_keys: keys,
+                row_ranges: vec![],
+            }),
+            ..ReadRowsRequest::default()
+        };
+        let mut result = vec![];
+        for (_, cells) in self.read_rows(request).await? {
+            for (_, cell_value) in cells {
+                result.push(bcs::from_bytes(&cell_value)?);
```

Comment:
same thought here too. I can see where it would be valuable to get the raw bytes out vs deserialize in the instances where we're just acting as a proxy and don't want to pay the cost of deserializing the data